Maple2.Server.Game/Manager/UgcMarketManager.cs

```diff
@@ -0,0 +1,160 @@
+        items = db.GetUgcListingsByAccountId(session.AccountId);
+        sales = db.GetMySoldUgcListings(session.AccountId);
+        favoriteDesigners = new Dictionary<long, PlayerInfo>();
+        IList<long> designerList = session.Config.GetFavoriteDesigners();
+
+        foreach (long characterId in designerList) {
+            if (!session.PlayerInfo.GetOrFetch(characterId, out PlayerInfo? playerInfo)) {
+                continue;
+            }
+            favoriteDesigners.Add(characterId, playerInfo);
+        }
+    }
+
+    public void Load() {
+        session.Send(MeretMarketPacket.LoadPersonalListings(items.Values));
+    }
+
+    public void RefreshSales() {
+        using GameStorage.Request db = session.GameStorage.Context();
+        Save(db);
+        sales.Clear();
+        sales = db.GetMySoldUgcListings(session.AccountId);
+        session.Send(MeretMarketPacket.LoadSales(sales.Values));
+    }
+
+    public void Add(UgcMarketItem item) {
+        items.Add(item.Id, item);
+        session.Send(MeretMarketPacket.ListItem(item));
+        session.Send(MeretMarketPacket.UpdateExpiration(item));
+    }
+
+    public void CollectProfit(long id) {
+        if (!sales.TryGetValue(id, out SoldUgcMarketItem? sale) ||
+            sale.SoldTime.FromEpochSeconds().AddDays(Constant.UGCShopProfitDelayDays) > DateTime.Now) {
+            return;
+        }
+        using GameStorage.Request db = session.GameStorage.Context();
+        if (!db.DeleteSoldUgcMarketItem(id)) {
+            logger.Error("Unable to collect profit and delete SoldUgcMarketItem {Id}.", id);
+            return;
+        }
+        sales.Remove(id);
+
+        session.Currency.Meret += sale.Profit;
+        session.Send(MeretMarketPacket.UpdateProfit(sale.Id));
+    }
+
+    public void RemoveListing(long id) {
+        if (!items.ContainsKey(id)) {
+            return;
+        }
+
+        using GameStorage.Request db = session.GameStorage.Context();
+        if (!db.DeleteUgcMarketItem(id)) {
+            return;
+        }
+
+        items.Remove(id);
+        session.Send(MeretMarketPacket.RemoveListing(id));
+    }
+
+    public void UnlistItem(long id) {
+        if (!items.TryGetValue(id, out UgcMarketItem? item)) {
+            return;
+        }
+
+        item.ListingEndTime = 0;
+        item.PromotionEndTime = 0;
+        item.Status = UgcMarketListingStatus.Expired;
+
+        using GameStorage.Request db = session.GameStorage.Context();
+        if (!db.SaveUgcMarketItem(item)) {
+            return;
+        }
+
+        session.Send(MeretMarketPacket.UpdateExpiration(item));
+    }
+
+    public void AddDesigner(long characterId) {
+        if (favoriteDesigners.ContainsKey(characterId)) {
+            LoadDesignerItems(characterId);
+            return;
+        }
+
+        using GameStorage.Request db = session.GameStorage.Context();
+        PlayerInfo? info = db.GetPlayerInfo(characterId);
```

Comment:
Avoid reading `PlayerInfo` directly from db, use `session.PlayerInfo.GetOrFetch(...)` so it's cached (and less stale)